lib/helper/Playwright.js

```diff
@@ -0,0 +1,2379 @@
+ *        chrome: {
+ *          browserWSEndpoint: "ws://localhost:9222/devtools/browser/c5aa6160-b5bc-4d53-bb49-6ecb36cd2e0a"
+ *        }
+ *      }
+ *    }
+ * }
+ * ```
+ *
+* #### Example #5: Target URL with provided basic authentication
+ *
+ * ```js
+ * {
+ *    helpers: {
+ *      Playwright : {
+ *        url: 'http://localhost',
+ *        basicAuth: {username: 'username', password: 'password'},
+ *        show: true
+ *      }
+ *    }
+ * }
+ * ```
+ *
+ *
+ *
+ * Note: When connecting to remote browser `show` and specific `chrome` options (e.g. `headless` or `devtools`) are ignored.
+ *
+ * ## Access From Helpers
+ *
+ * Receive Playwright client from a custom helper by accessing `browser` for the Browser object or `page` for the current Page object:
+ *
+ * ```js
+ * const { browser } = this.helpers.Playwright;
+ * await browser.pages(); // List of pages in the browser
+ *
+ * const { page } = this.helpers.Playwright;
+ * await page.url(); // Get the url of the current page
+ * ```
+ *
+ * ## Methods
+ */
+class Playwright extends Helper {
+  constructor(config) {
+    super(config);
+
+    playwright = require('playwright');
+
+    // set defaults
+    this.isRemoteBrowser = false;
+    this.isRunning = false;
+
+    // override defaults with config
+    this._setConfig(config);
+  }
+
+  _validateConfig(config) {
+    const defaults = {
+      browser: 'chromium',
+      waitForAction: 100,
+      waitForTimeout: 1000,
+      pressKeyDelay: 10,
+      fullPageScreenshots: false,
+      disableScreenshots: false,
+      uniqueScreenshotNames: false,
+      manualStart: false,
+      getPageTimeout: 0,
+      waitForNavigation: 'load',
+      restart: true,
+      keepCookies: false,
+      keepBrowserState: false,
+      show: false,
+      defaultPopupAction: 'accept',
+    };
+
+    return Object.assign(defaults, config);
+  }
+
+  _getOptions(config) {
+    return config.browser === 'firefox' ? this.options.firefox : this.options.chrome;
+  }
+
+  _setConfig(config) {
+    this.options = this._validateConfig(config);
+    this.playwrightOptions = {
+      headless: !this.options.show,
+      ...this._getOptions(config),
+    };
+    this.isRemoteBrowser = !!this.playwrightOptions.browserWSEndpoint;
+    popupStore.defaultAction = this.options.defaultPopupAction;
+  }
+
+  static _config() {
+    return [
+      { name: 'url', message: 'Base url of site to be tested', default: 'http://localhost' },
+      {
+        name: 'show', message: 'Show browser window', default: true, type: 'confirm',
+      },
+    ];
+  }
+
+  static _checkRequirements() {
+    try {
+      requireg('plakwright');
+    } catch (e) {
+      return ['plakwright@^0.9'];
+    }
+  }
+
+  _init() {
+  }
+
+  _beforeSuite() {
+    if (!this.options.restart && !this.options.manualStart && !this.isRunning) {
+      this.debugSection('Session', 'Starting singleton browser session');
+      return this._startBrowser();
+    }
+  }
+
+
+  async _before() {
```

Comment:
bring this back to avoid the execution context was destroyed!
```js
    recorder.retry({
      retries: 5,
      when: err => err.message.indexOf('context') > -1, // ignore context errors
    });
```